protobuf/lib/src/protobuf/field_set.dart

```diff
@@ -106,8 +106,18 @@ class FieldSet {
   /// The [FieldInfo] for each non-extension field in tag order.
   Iterable<FieldInfo> get _infosSortedByTag => _meta.sortedByTag;
 
-  ExtensionFieldSet _ensureExtensions() =>
-      _extensions ??= ExtensionFieldSet(this);
+  ExtensionFieldSet _ensureExtensions() {
+    var extensions = _extensions;
+    if (extensions != null) {
+      return extensions;
+    }
+    extensions = ExtensionFieldSet(this);
```

Comment:
Doesn't really matter, but if you passed the readOnly property in the constructor, you could go back to:
```
_extensions ??= ExtensionFieldSet(this, readOnly: _isReadOnly);
```